plugin/rename_file.py

```diff
@@ -0,0 +1,154 @@
+        return True
+
+    def run(self, paths: list[str] | None = None) -> None:
+        old_path = paths[0] if paths else None
+        if old_path:
+            self.window.run_command('lsp_rename_path', {
+                "old_path": old_path
+            })
+
+
+class RenamePathInputHandler(sublime_plugin.TextInputHandler):
+    def __init__(self, path: str) -> None:
+        self.path = path
+
+    def name(self) -> str:
+        return "new_name"
+
+    def placeholder(self) -> str:
+        return self.path
+
+    def initial_text(self) -> str:
+        return self.placeholder()
+
+    def initial_selection(self) -> list[tuple[int, int]]:
+        name = Path(self.path).stem
+        return [(0, len(name))]
+
+    def validate(self, path: str) -> bool:
+        return len(path) > 0
+
+
+class LspRenamePathCommand(LspWindowCommand):
+    capability = 'workspace.fileOperations.willRename'
+
+    def is_enabled(self) -> bool:
+        return True
+
+    def want_event(self) -> bool:
+        return False
+
+    def input(self, args: dict) -> sublime_plugin.TextInputHandler | None:
+        if "new_name" in args:
+            return None
+        view = self.window.active_view()
+        old_path = args.get('old_path') or view.file_name() if view else None
+        return RenamePathInputHandler(Path(old_path or "").name)
+
+    def run(self, new_name: str, old_path: str | None = None) -> None:
+        view = self.window.active_view()
+        old_path = old_path or view.file_name() if view else None
+        if old_path is None:  # handle renaming buffers
+            if view:
+                view.set_name(new_name)
+            return
+        if new_name == old_path:
+            return
+        # new_name can be: FILE_NAME.xy OR ./FILE_NAME.xy OR ../../FILE_NAME.xy
+        resolved_new_path = (Path(old_path).parent / new_name).resolve()
+        new_path = str(resolved_new_path)
+        if resolved_new_path.exists() and not self.is_case_change(old_path, new_path):
+            self.window.status_message('Unable to Rename. Already exists')
+            return
+        sublime.set_timeout_async(lambda: self.run_async(old_path, new_path))
+
+    def run_async(self, old_path: str, new_path: str) -> None:
+        file_rename: FileRename = {
+            "newUri": filename_to_uri(new_path),
+            "oldUri": filename_to_uri(old_path)
+        }
+        promises = list(self.create_will_rename_requests_async(file_rename))
+        if promises:
+            Promise.all(promises).then(lambda responses: self.handle_rename_async(responses, file_rename))
+        else:
+            self.rename_path(file_rename)
+
+    def create_will_rename_requests_async(
+        self, file_rename: FileRename
+    ) -> Generator[Promise[tuple[WorkspaceEdit | None, weakref.ref[Session]]]]:
+        for session in self.sessions():
+            filters = session.get_capability('workspace.fileOperations.willRename.filters') or []
```

Comment:
Just for the record I will say that running this for all sessions can be both good and bad. :)

With Vue files it could be beneficial while for other cases, when running two competing servers, it could cause conflicts.

I probably lean more towards running for all servers and making the user responsible for not running conflicting servers at the same time or disabling capabilities manually.